core/src/main/java/org/opensearch/sql/analysis/AnalysisContext.java

```diff
@@ -23,13 +27,27 @@ public class AnalysisContext {
   @Getter
   private final List<NamedExpression> namedParseExpressions;
 
+  /**
+   * Storage for values of functions which return a constant value.
+   * We are storing the values there to use it in sequential calls to those functions.
+   * For example, `now` function should the same value during processing a query.
```

Comment:
I'm open to rename it.

Regarding arguments - they are already included, **but** `now` function (and all synonyms) doesn't accept argument anymore. Otherwise there is bug, when `now()` and `now(6)` have different values, because were calculated twice, because they can't cross-use cached value.

Reply:
Yes.  No arguments on `now` is fine.

Reply:
Renaming done in ce2e8bfc.